Uat dpl258 add sufix checksum (#258)
sanger/baracoda: README.md

If true, the prefix will support creating barcodes based on a parent barcode
If it is False, the prefix will reject any children creation request for that prefix.

### Formatter classes

A formatter class is a way of rendering a new barcode created that can be attached to a prefix

> through the configuration before. Any new Formatter class must extend from the abstract class

Reviewer: Not sure what `through the configuration before` means

```baracoda.formats.FormatterInterface```.

The current list of supported formatter classes that can be specified for a prefix is:

- HeronCogUkIdFormatter
- GenericBarcodeFormatter
- Sequencescape22Formatter

#### HeronCogUkIdFormatter

Implements the Heron COG UK Id generation that corresponds with format: ```<Prefix>-<Hexadecimal><Checksum>```.
Examples: ```ASDF-A34ADA```

With description:

- Prefix is the prefix provided to the formatter.
- Hexadecimal is the hexadecimal representation for the current index for the id.
- Checksum is a checksum hexadecimal character that validates the hexadecimal value.

#### GenericBarcodeFormatter

Standard barcode generator that follows the format: ```<Prefix>-<Number>```.
Examples: ```HT-11811```

With description:

- Prefix is the prefix provided to the formatter
- Number is the current index for the id of this barcode.

#### Sequencescape22Formatter

Plate barcode generator which supports children creation and checksum. It follows these 2 formats:

>  - For any barcode that is not a children: ```<Prefix>-<Number>-<Checksum>```. Examples: ```SQDP-23-L```

Reviewer: ?
```suggestion
 - For any barcode that is not a child: ```<Prefix>-<Number>-<Checksum>```. Examples: ```SQDP-23-L```
```

>  - For any children barcode: ```<Prefix>-<Number>-<ChildIndex>-<Checksum>```. Examples: ```LAB-89-2-R```

Reviewer: ```suggestion
 - For any child barcode: ```<Prefix>-<Number>-<ChildIndex>-<Checksum>```. Examples: ```LAB-89-2-R```
```

Author: Ok

extracted from the URL ```/child-barcodes/<PREFIX>/new```
All barcodes for the children will have this prefix (example, prefix SS will generate children
like SS-11111-1-L, SS-11111-2-M, etc)
- *Parent Barcode* : barcode that will act as parent of the children. This argument will be
extracted from the Body of the request, eg: ```{'barcode': 'SS-1-1-L', 'count': 2}```.
To be considered valid, the barcode needs to follow the format ```<PREFIX>-<NUMBER>(-<NUMBER>)?(-<CHECKSUM>)```
where the second number part is optional and it represents if the barcode was a child.
For example, valid barcodes would be ```SS-11111-13-M``` (normal parent) and ```SS-11112-24-N```
(parent that was a child) but not ```SS-1-1-1-L``` (several '-') or ```SS12341-1-L``` (no '-' separation).
- *Child* : part of the Parent barcode string that would identify if the parent was
a child before (the last number). For example for the *Parent barcode* ```SS-11111-14-R```,
*Child* would be 14; but for the *Parent barcode* ```SS-11111-W```, *Child* would have no
value defined.
- *Count* : number of children barcodes to create.

#### Wrong parent barcodes

> A request with parent barcode that does not follow the format defined like:

Reviewer: ```suggestion
A request with parent barcode that does not follow the format defined by `Sequencescape22Formatter`
```

```

> <PREFIX>-<NUMBER>(-<NUMBER>)?(-<CHECKSUM>)

Reviewer: Is this the same for all formats?

Author: Only for formats that accept children barcodes

```
will create normal barcodes instead of suffixed children barcodes (normal barcode creation).

A request that follows the right format but does not comply with current database will be

> rejected as impostor barcode. For example, if we receive the parent barcode ```SS-1111-14-N```, but in the

Reviewer: ```suggestion
rejected as an impostor barcode. For example, if we receive the parent barcode ```SS-1111-14-N```, but in the
```
